onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -1243,7 +1243,7 @@ common::Status TensorrtExecutionProvider::Compile(const std::vector<Node*>& fuse
             &engines_[context->node_name], &contexts_[context->node_name], &builders_[context->node_name],
             &networks_[context->node_name], input_info_[context->node_name], output_info_[context->node_name],
             input_shape_ranges_[context->node_name], &tensorrt_mu_, &fp16_enable_, &int8_enable_, &max_workspace_size_,
-            trt_node_name_with_precision, engine_cache_enable_, cache_path_, runtime_, nullptr,
+            trt_node_name_with_precision, engine_cache_enable_, cache_path_, &runtime_, nullptr,
```

Comment:
what if we just pass the raw pointer here instead, instead of passing the address of the unique_ptr

Reply:
Yes, we can pass the raw pointer as before. But we need to call runtime_.destropy() in destructor in order to release IRuntime object.

Previously, I'm thinking about passing unique_ptr, but it will have following compile error:
/home/onnxruntime/repos/onnxruntime/onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc:1247:89: error: use of deleted function ‘std::unique_ptr<_Tp, _Dp>::unique_ptr(const std::unique_ptr<_Tp, _Dp>&) [with _Tp = nvinfer1::IRuntime; _Dp = onnxruntime::tensorrt_ptr::TensorrtInferDeleter]’

The reason is that Copy constructor and Copy assignment of unique_ptr are deleted functions. So it will get error on line 1246. I think that's the reason that in `struct TensorrtFuncState`, we pass pointer to unique_ptr, such as parser, engine, context....

Reply:
runtime is already wrapped with a unique_ptr in TrtExecutionProvider right?
it will get freed when TrtExecutionProvider gets destroyed.
what you did, passing address of unique_ptr to the TensorrtFuncState is basically like passing in the underlying raw pointer, no?

Reply:
I think passing raw pointer is also another solution, but it will lose code consistency since other members in `struct TensorrtFuncState` uses the defined unique_pointer which will handle destroy() automacially.

Reply:
discussed offline. need to clean up some of this code.